src/server/api/services/EventProcessor.ts

```diff
@@ -325,6 +325,7 @@ export class TraceProcessor implements EventProcessor {
           : undefined,
         public: body.public ?? undefined,
         project: { connect: { id: apiScope.projectId } },
+        tags: body.tags ?? undefined,
```

Comment:
Should we use undefined here? I would expect that we always have an array, empty or filled.

Reply:
Does this work? According to the prisma schema, tags cannot be null.

Reply:
From my understanding: If there are no tags present in body object the function should not upsert any tags at all. If there are no tags present in the body object it triggers the default of the prisma schema, adding an empty array to the trace which is the behaviour we want.
It gives the upsert and create function more flexibility for the case that that the body does not contain tags.

Can assume that every trace body when upserted or created will always have tags in it?